### PR TITLE
goal node create improvements

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

1. `--full-config` options allows saving config file with all options, not only non-default.
2. Embed genesis.json for mainnet, testnet, betanet, devnet

## Test Plan

Tested manually

### CODE REVIEW

**Review by jasonpaulos — DISMISSED** [review]
Changes make sense to me

File: cmd/goal/node.go [hunk 1]
```diff
@@ -595,6 +597,10 @@ var createCmd = &cobra.Command{
 		secondChoicePath := filepath.Join("var", "lib", "algorand", "genesis", newNodeNetwork, "genesis.json")
 		thirdChoicePath := filepath.Join(exePath, "genesisfiles", "genesis", newNodeNetwork, "genesis.json")
 		paths := []string{firstChoicePath, secondChoicePath, thirdChoicePath}
+		if gopath := os.Getenv("GOPATH"); len(gopath) > 0 {
+			fourthChoicePath := filepath.Join(gopath, "src", "github.com", "algorand", "go-algorand", "installer", "genesis", newNodeNetwork, "genesis.json")
+			paths = append(paths, fourthChoicePath)
+		}
```
Comment (tsachiherman) [inline cmd/goal/node.go:603]: I'm not a big fan of this approach. The goal should come with pre-baked knowledge of the 3 supported networks, plus the ability to specify a custom genesis file. There shouldn't be any "search" done.

Comment (algorandskiy) [reply cmd/goal/node.go:603]: I kind of agree, the GOPATH approach not ideal but at least it makes `goal node create` working on developers machines why do not use updater/packages.

**Comment by algorandskiy** [comment]
@tsachiherman followed your suggestion and embedded genesis files.

File: cmd/goal/bundle_genesis_json.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+THISDIR=$(dirname $0)
+
+cat <<EOM | gofmt > $THISDIR/bundledGenesisInject.go
+// Code generated by bundle_genesis_json.sh, along with langspec.json. DO NOT EDIT.
```
Comment (jasonpaulos) [inline cmd/goal/bundle_genesis_json.sh:6]: I don't think langspec.json is relevant here

File: cmd/goal/bundledGenesisInject.go [hunk 3]
```diff
@@ -0,0 +1,2793 @@
+// Code generated by bundle_genesis_json.sh, along with langspec.json. DO NOT EDIT.
```
Comment (winder) [inline cmd/goal/bundledGenesisInject.go:1]: so close but so far: https://pkg.go.dev/embed

File: cmd/goal/node.go [hunk 4]
```diff
@@ -92,6 +96,7 @@ func init() {
 	createCmd.Flags().BoolVarP(&newNodeIndexer, "indexer", "i", localDefaults.IsIndexerActive, "Configure the new node to enable the indexer feature (implies --archival)")
 	createCmd.Flags().StringVar(&newNodeRelay, "relay", localDefaults.NetAddress, "Configure as a relay with specified listening address (NetAddress)")
 	createCmd.Flags().StringVar(&listenIP, "api", "", "REST API Endpoint")
+	createCmd.Flags().BoolVar(&newNodeFullConfig, "full-config", false, "Store full config file")
```
Comment (winder) [inline cmd/goal/node.go:99]: Should this default to true?

**Review by winder — APPROVED** [review]
LGTM

File: cmd/goal/bundledGenesisInject.go (hunk 3 above)
Comment (algorandskiy) [reply cmd/goal/bundledGenesisInject.go:1]: it is only for 1.16+ folks, not for us

File: cmd/goal/node.go (hunk 4 above)
Comment (algorandskiy) [reply cmd/goal/node.go:99]: I preserve the old behavior by default, mb some folks like it

File: cmd/goal/bundle_genesis_json.sh (hunk 2 above)
Comment (algorandskiy) [reply cmd/goal/bundle_genesis_json.sh]: fixed

**Review by jasonpaulos — APPROVED** [review]

**Comment by algorandskiy** [comment]
Closes #1686

**Review by tsachiherman — APPROVED** [review]
looks good.

**Pull request merged**